### PR TITLE
Fix up log level from env var

### PR DESCRIPTION
**Pull request by caseydavenport**
- Correctly set log severity from env var at start of day. 
- Remove support for the cli flag `--log-level`
- Use logrus directly, so the correct line numbers and file names appear in log output.

### CODE REVIEW

File: log/log.go [hunk 1]
```diff
@@ -41,28 +40,3 @@ func SetLevel(level string) {
 	}
 	log.SetLevel(lvl)
 }
-
```
Comment (robbrockbank) [inline log/log.go]: I admit, I'm not sad to see this go.

File: resource/template/resource.go [hunk 2]
```diff
@@ -13,9 +13,10 @@ import (
 	"strings"
 	"text/template"
 
+	log "github.com/sirupsen/logrus"
+
```
Comment (robbrockbank) [inline resource/template/resource.go]: nit: extra line

File: log/log.go (hunk 1 above)
Comment (robbrockbank) [reply log/log.go]: Not sure if we need the SetTag anymore.

File: tests/test_suite_common.sh [hunk 3]
```diff
@@ -43,7 +43,7 @@ execute_test_suite() {
 execute_tests_daemon() {
     # Run confd as a background process.
     echo "Running confd as background process"
-    confd -confdir=/etc/calico/confd -log-level=debug >$LOGPATH/logd1 2>&1 &
+    BGP_LOGSEVERITYSCREEN="debug" confd -confdir=/etc/calico/confd >$LOGPATH/logd1 2>&1 &
```
Comment (robbrockbank) [inline tests/test_suite_common.sh]: extremely minor nit:  do we need these quotes?

File: backends/calico/client.go [hunk 4]
```diff
@@ -7,7 +7,8 @@ import (
 	"strings"
 	"sync"
 
-	"github.com/kelseyhightower/confd/log"
+	logutils "github.com/kelseyhightower/confd/log"
```
Comment (robbrockbank) [inline backends/calico/client.go]: worth renaming the package to logutils?

File: tests/test_suite_common.sh (hunk 3 above)
Comment (caseydavenport) [reply tests/test_suite_common.sh]: We don't, but I've already typed them . . .

File: backends/calico/client.go (hunk 4 above)
Comment (caseydavenport) [reply backends/calico/client.go]: nah

File: backends/calico/client.go [hunk 5]
```diff
@@ -162,7 +163,11 @@ func (c *client) OnStatusUpdated(status api.SyncStatus) {
 		defer c.cacheLock.Unlock()
 		c.synced = true
 		c.waitForSync.Done()
-		log.Debug("Data is now syncd, can start rendering templates")
+		log.Debugf("Data is now syncd, can start rendering templates")
+
+		// Now that we're in-sync, check if we should update our log level
```
Comment (caseydavenport) [inline backends/calico/client.go]: @robbrockbank we were missing this before - this lets us set log level when we connect to the datastore for the first time.

Comment (robbrockbank) [reply backends/calico/client.go]: D'oh yes!

File: backends/calico/client.go [hunk 6]
```diff
@@ -162,7 +163,11 @@ func (c *client) OnStatusUpdated(status api.SyncStatus) {
 		defer c.cacheLock.Unlock()
 		c.synced = true
 		c.waitForSync.Done()
-		log.Debug("Data is now syncd, can start rendering templates")
+		log.Debugf("Data is now syncd, can start rendering templates")
```
Comment (robbrockbank) [inline backends/calico/client.go]: nit:  don't really need Debugf here, but doesn't really matter I suppose.

**Comment by caseydavenport** [comment]
@robbrockbank there are three distinct changes in this PR, so I think the three commits are good as they are.

**Comment by caseydavenport** [comment]
Eh, nevermind. I'll squash anyway.

**Pull request merged**